Enablement flag clean-ups

brentfulgham opened the pull request:
#### 8844e5028d10063aaf5ff4d20c3e96906ca7a18c
<pre>
Enablement flag clean-ups
<a href="https://bugs.webkit.org/show_bug.cgi?id=261843">https://bugs.webkit.org/show_bug.cgi?id=261843</a>
&lt;rdar://problem/115807440&gt;

Reviewed by Tim Horton and Elliott Williams.

This patch moves a few more enablement flags into Open Source, and corrects
some mistakes where some HAVE_ macros were placed in ENABLE_ header files.

* Source/WTF/wtf/PlatformEnable.h:
- Add WEBXR enablement flags for visionOS.
- Move a Cocoa-specific feature flag to PlatformEnableCocoa.h
* Source/WTF/wtf/PlatformEnableCocoa.h:
- Move some visionOS HAVE_ macros to PlatformHave.h.
- Move some visionOS ENABLE_ macros to this Open Source file.
- Add ENABLE_IMAGE_ANALYSIS_ENHANCEMENTS flag from PlatformEnable.h
* Source/WTF/wtf/PlatformHave.h:
- Add HAVE_UI_FOCUS_EFFECT to Open Source.
- Add visionOS HAVE_ macros previously in PlatformEnableCocoa.h
* Source/WebCore/platform/ios/UserAgentIOS.mm:
- Switch definition to ENABLE_STATIC_IPAD_USER_AGENT_VALUE, and move to Open Source.

Canonical link: <a href="https://commits.webkit.org/268299@main">https://commits.webkit.org/268299@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ba8ee21cb44d0685c7042fb1f96c60397f1ec8dc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/19226 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/19641 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/20239 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/21115 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/17989 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/19442 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/22912 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/19767 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/19661 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/19447 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/19516 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/16722 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/21985 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/16705 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/17508 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/23859 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/16686 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/17764 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/17683 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/21818 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/18553 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/18267 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/15471 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/22616 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/17396 "Built successfully") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/5497 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4606 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/21753 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/23866 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/18102 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/5337 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WTF/wtf/PlatformEnable.h:
```diff
@@ -549,12 +549,20 @@
 #endif
 
 #if !defined(ENABLE_WEBXR)
+#if PLATFORM(VISION)
```
PlatformEnableCocoa?

brentfulgham replied:
I figured other ports might enable WebXR at some point, so it wasn't really a cocoa-specific flag.

brentfulgham replied:
Oh! I misunderstood how these layer. Yes, I'll move it.

hortont424 replied:
Right, this is the default value for WebKit as a whole, and the platforms override in their children

hortont424 commented on Source/WTF/wtf/PlatformEnable.h:
```diff
@@ -549,12 +549,20 @@
+#endif
 
 #if !defined(ENABLE_WEBXR_HANDS)
+#if PLATFORM(VISION)
```
Same.

hortont424 commented on Source/WTF/wtf/PlatformEnableCocoa.h:
```diff
@@ -800,8 +794,17 @@
-#if !defined(ENABLE_APP_HIGHLIGHTS) \
-    && (PLATFORM(MAC) || PLATFORM(IOS) || PLATFORM(VISION))
+#if PLATFORM(VISION)
+#define ENABLE_APP_HIGHLIGHTS 0
```
Now that we're upstreaming these, we should make them conditions on the #defines that enable the feature, not have this fun/unusual mass-override, right?

hortont424 commented on Source/WTF/wtf/PlatformUse.h:
```diff
@@ -157,6 +157,10 @@
 #endif
 
+#if PLATFORM(VISION)
+#define USE_STATIC_IPAD_USER_AGENT_VALUE 1
```
I know this predates you, but this is an ENABLE, NOT a USE ("use a particular third-party library or optional OS service")

brentfulgham replied:
Oh, I could move it to the PlatformEnableCocoa header, and if it's only used in one or two places maybe change it to ENABLE_STATIC_IPAD_USER_AGENT_VALUE.

brentfulgham committed:
Enablement flag clean-ups
https://bugs.webkit.org/show_bug.cgi?id=261843
<rdar://problem/115807440>

Reviewed by Tim Horton and Elliott Williams.

This patch moves a few more enablement flags into Open Source, and corrects
some mistakes where some HAVE_ macros were placed in ENABLE_ header files.

* Source/WTF/wtf/PlatformEnable.h:
- Add WEBXR enablement flags for visionOS.
- Move a Cocoa-specific feature flag to PlatformEnableCocoa.h
* Source/WTF/wtf/PlatformEnableCocoa.h:
- Move some visionOS HAVE_ macros to PlatformHave.h.
- Move some visionOS ENABLE_ macros to this Open Source file.
- Add ENABLE_IMAGE_ANALYSIS_ENHANCEMENTS flag from PlatformEnable.h
* Source/WTF/wtf/PlatformHave.h:
- Add HAVE_UI_FOCUS_EFFECT to Open Source.
- Add visionOS HAVE_ macros previously in PlatformEnableCocoa.h
* Source/WebCore/platform/ios/UserAgentIOS.mm:
- Switch definition to ENABLE_STATIC_IPAD_USER_AGENT_VALUE, and move to Open Source.

Canonical link: https://commits.webkit.org/268299@main
